app/src/main/java/com/guillermorosales/smarthome/MainActivity.kt

```diff
@@ -11,16 +17,17 @@ class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-        switch_lamp.setOnCheckedChangeListener {_, checked -> doSomething(checked)}
+        bindViewModel()
     }
 
-    private fun doSomething(checked: Boolean) {
-        // Write a message to the database
-        val database = FirebaseDatabase.getInstance()
-        val myRef = database.reference
-
-        myRef.child("lights").child("on").setValue(checked)
-
+    private fun bindViewModel() {
+        val model = ViewModelProviders.of(this).get(MainViewModel::class.java)
+        model.getLightState().observe(this, Observer<Boolean> { state ->
+            switch_lamp.isChecked = state!!
+        })
 
+        RxCompoundButton.checkedChanges(switch_lamp).subscribe {
```

Comment:
qq, what's the advantage of using this versus the setOnCheckedChangeListener?

Reply:
This is the reactive way, you avoid checking nullability and instead of using a callback you subscribe to the an observable that the RxBindingis exposing so you can use streams operators and  other advantages, check this post.
https://robots.thoughtbot.com/observables-and-callbacks-and-threading-oh-my